numerizer/numerizer.py

```diff
@@ -237,17 +237,29 @@ def postprocess(s, ignore=None):
 
 
 def andition(s):
-    pat = re.compile(r'<num>(\d+)( | and )<num>(\d+)(?=[^\w]|$)', flags=re.IGNORECASE)
+    # Define a pattern for finding the numbers to be unified
+    pattern = re.compile(r'<num>(\d+)( | and )<num>(\d+)(?=[^\w]|$)', flags=re.IGNORECASE)
+
     while True:
-        m = re.search(pat, s)
-        if m is not None:
-            if (m.group(2) == 'and') or (len(m.group(1)) > len(m.group(3))):
-                s = re.sub(pat, lambda m: '<num>' + str(int(m.group(1)) + int(m.group(3))),
-                           s, count=1)
-            else:
-                break
+        matches = list(re.finditer(pattern, s))
```

Comment:
Instead of creating an iterator first through `re.finditer` and then converting it into a list, shall we use `re.findall(pattern, s)`? That removes the unnecessary `list`ifying of a generator.

Reply:
That's a good point - we need match objects, not just the matched substrings.